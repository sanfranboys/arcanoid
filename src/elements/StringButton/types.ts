export type StringButtonProps = {
  onClick?: () => void
}
