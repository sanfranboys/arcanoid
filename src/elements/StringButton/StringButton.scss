.string-button {
  font-weight: bold;
  color: var(--link);
  border: none;
  background: none;

  &:hover {
    cursor: pointer;
    color: var(--link-hover);
  }
}
