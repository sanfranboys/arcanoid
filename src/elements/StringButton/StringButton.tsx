import React, { FC } from 'react'
import { StringButtonProps } from './types'

import './StringButton.scss'

const StringButton: FC<StringButtonProps> = ({
  children,
  onClick = () => {},
}) => (
  <span className="string-button" onClick={onClick} onKeyDown={onClick}>
    {children}
  </span>
)

export default StringButton
