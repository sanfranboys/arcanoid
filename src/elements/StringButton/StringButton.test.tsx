import React from 'react'
import { shallow, ShallowWrapper } from 'enzyme'
import { create } from 'react-test-renderer'
import StringButton from './StringButton'

describe('StringButton testing', () => {
  let wrapper: ShallowWrapper
  const props = {
    children: 'Push me',
    onClick: () => {
      /* eslint-disable no-alert */
      alert('StringButton was clicked')
    },
  }

  beforeEach(() => {
    wrapper = shallow(<StringButton {...props} />)
  })

  describe('StringButton snapshot testing', () => {
    it('snapshot matches', async () => {
      const tree = create(<StringButton {...props} />).toJSON()
      expect(tree).toMatchSnapshot()
    })
  })

  describe('StringButton event testing', () => {
    it('StringButton handleClick works', async () => {
      window.alert = jest.fn()
      wrapper.simulate('click')
      expect(window.alert).toHaveBeenCalledWith('StringButton was clicked')
    })

    it('StringButton handleKeyDown works', async () => {
      window.alert = jest.fn()
      wrapper.simulate('keydown')
      expect(window.alert).toHaveBeenCalledWith('StringButton was clicked')
    })
  })
})
