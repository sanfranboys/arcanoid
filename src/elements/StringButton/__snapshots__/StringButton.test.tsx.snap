// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StringButton testing StringButton snapshot testing snapshot matches 1`] = `
<button
  className="string-button"
  onClick={[Function]}
  onKeyDown={[Function]}
  type="button"
>
  Push me
</button>
`;
